### PR TITLE
Add system tests for CLI

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

#2142

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: system-tests/cli/cli.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+"use strict";
```
Comment (evilebottnawi) [inline system-tests/cli/cli.test.js]: Also to start process we can use `execa`

**Comment by gucong3000** [comment]
When I used jest instead of sinon, I got an error:
```shell
(node:29688) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unhandledRejection listeners added. Use emitter.setMaxListeners() to increase limit
```

system-tests/cli/cli.test.js
```js
/* eslint no-console: off */
"use strict";
const cli = require("../../lib/cli");
const binPath = require.resolve("../../bin/stylelint")

describe("cli", () => {
  let processRestore;
  let logRestore;

  beforeAll(() => {
    processRestore = Object.assign({}, process);
    logRestore = Object.assign({}, console);
    process.exit = jest.fn();
    console.log = jest.fn();
  });

  afterAll(() => {
    Object.assign(process, processRestore);
    Object.assign(console, logRestore);
  });

  beforeEach(function() {
    process.argv = processRestore.argv;
  });

  it("basic", () => {
    process.argv = ["node", binPath];
    return cli().then(() => {
      expect(process.exit.mock.calls[0][0]).toBe(0);
    });
  });
});
```

**Comment by gucong3000** [comment]
> Also to start process we can use execa

That's what `system-tests/004/004.test.js` did, but jest doesn't seem to be able to statistically correlate coverage reporting information.

**Comment by jeddy3** [comment]
It'd be ideal if this was achieved with jest. We had previously used sinon but removed it when we migrated to jest. We did this to keep the tests consistent.

**Comment by CAYdenberg** [comment]
@gucong3000 I don't get that Jest error with this branch, either because I'm using a different version of Node (7.7.3) or Jest has been bumped since this was last addressed (21.2.1).

Do you want to rebase your branch (not trivial, since lib/cli.js has changed) and see if we can get this merged in?

File: system-tests/cli/cli.test.js (hunk 1 above)
Comment (gucong3000) [reply system-tests/cli/cli.test.js]: OK, I'll use `execa` if necessary.

**Comment by gucong3000** [comment]
@CAYdenberg After rebase, the test passed again.

**Comment by gucong3000** [comment]
All Done.

File: system-tests/cli/cli.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    return cli([]).then(() => {
+      expect(process.exitCode).toBe(2);
+      expect(console.log.mock.calls).toHaveLength(1);
+      const lastCallArgs = console.log.mock.calls.pop();
+      expect(lastCallArgs).toHaveLength(1);
+      expect(lastCallArgs.pop()).toMatch("Usage: stylelint [input] [options]");
+    });
+  });
+
+  it.skip("--help", () => {
```
Comment (jeddy3) [inline system-tests/cli/cli.test.js]: Should this tests be skipped?

File: system-tests/cli/cli.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    return Promise.resolve(cli(["--help"])).then(() => {
+      expect(process.exitCode).toBe(0);
+      expect(console.log.mock.calls).toHaveLength(1);
+      const lastCallArgs = console.log.mock.calls.pop();
+      expect(lastCallArgs).toHaveLength(1);
+      expect(lastCallArgs.pop()).toMatch("Usage: stylelint [input] [options]");
+    });
+  });
+
+  it.skip("--version", () => {
```
Comment (jeddy3) [inline system-tests/cli/cli.test.js]: Should this tests be skipped?

Comment (gucong3000) [reply system-tests/cli/cli.test.js]: Sorry, it was my fault. fixed.

**Comment by jeddy3** [comment]
After a second review on this, I think we can merge.

**Comment by gucong3000** [comment]
> Please, run npm run prettier:fix.

Done.

**Pull request merged**